cli/flox/src/commands/environment.rs

```diff
@@ -570,25 +612,171 @@ impl Activate {
             },
         };
 
-        if !self.run_args.is_empty() {
+        if !run_args.is_empty() {
             command.arg("-i");
             command.arg("-c");
-            command.arg(self.run_args.join(" "));
+            command.arg(run_args.join(" "));
         }
 
         debug!("running activation command: {:?}", command);
 
-        if self.run_args.is_empty() {
+        if run_args.is_empty() {
             let message = formatdoc! {"
                 ✅  You are now using the environment {now_active}.
                 To stop using this environment, type 'exit'"};
             info!("{message}");
         }
-        let error = command.exec();
-
         // exec should never return
+        command.exec().into()
+    }
+
+    /// Patch the PATH to undo the effects of `/usr/libexec/path_helper`
+    ///
+    /// Darwin has a "path_helper" which indiscriminately reorders the path
+    /// to put the system curated path items first in the `PATH`,
+    /// which completely breaks the user's ability to manage their `PATH` in subshells,
+    /// e.g. when using tmux.
+    ///
+    /// On macos `/usr/libexec/path_helper` is typically invoked from
+    /// default shell rc files, e.g. `/etc/profile` and `/etc/zprofile`.
+    ///
+    /// This function attempts to undo the effects of `/usr/libexec/path_helper`
+    /// by partitioning the `PATH` into two parts:
+    /// 1. known paths of activated flox environments
+    ///    and nix store paths (e.g. `/nix/store/...`) -- assumed to be `nix develop` paths
+    /// 2. everything else
+    ///
+    /// The `PATH` is then reordered to put the flox environment and nix store paths first.
+    /// The order within the two partitions is preserved.
+    fn fixup_path(flox_env_dirs: IndexSet<PathBuf>) -> Option<Vec<PathBuf>> {
+        if !Path::new("/usr/libexec/path_helper").exists() {
+            return None;
+        }
+        let path_var = env::var("PATH").unwrap_or_default();
+        Some(Self::fixup_path_with(path_var, flox_env_dirs))
+    }
```

Comment:
```suggestion
    #[cfg_attr(not(target_os = "macos"), allow(unused))]
    fn fixup_path(flox_env_dirs: IndexSet<PathBuf>) -> Option<Vec<PathBuf>> {
        #[cfg(not(target_os = "macos"))]
        {
            None
        }
        #[cfg(target_os = "macos")]
        {
            if !Path::new("/usr/libexec/path_helper").exists() {
                return None;
            }
            let path_var = env::var("PATH").unwrap_or_default();
            Some(Self::fixup_path_with(path_var, flox_env_dirs))
        }
    }
```

Reply:
On `linux` this will have compiler warnings that `flox_env_dirs` is unused

Reply:
I see, fixed